### PR TITLE
Implementing hybrid upwinding for total velocity schemes

### PR DESCRIPTION
**Pull request by jafranc**
Following up #2410, this PR aims at implementing simple version of Implicit Hybrid Upwinding (_IHU_). It implements ``IHUPhaseFlux`` equivalent to ``C1PPUPhaseFlux`` or ``PPUPhaseFlux``  located in ``IsothermalCompositionalMultiphaseFVMKernelUtilities.hpp``. It serves as a dispatch value in ``FaceBasedAssemblyKernelFactory::createAndLaunch()``.

some refs on IHU:

["Implicit Hybrid Upwinding for two-phase flow in heterogeneous porous media with buoyancy and capillarity"(Hamon et al,2018)](https://www.sciencedirect.com/science/article/pii/S0045782517306771)
["Consistent upwinding for sequential fully implicit multiscale compositional simulation" (Moncorge et al,2020)](https://link.springer.com/article/10.1007/s10596-019-09835-6)
["Implicit Hybrid Upwinding for two-phase flow in heterogeneous porous media with buoyancy and capillarity"(Hamon et al,2018)](https://www.sciencedirect.com/science/article/pii/S0045782517306771)

It also introduces ``UpwindHelpers`` namespace to scope helpers functions computing upwinded fractional flows and potential fluxes that are useful in IHU (and other schemes). Eventually, it introduces an Abstract class ``UpwindScheme`` introducing high level mechanisms to compute upwind direction and fluxes and its Derived class ``HybridUpwind`` to implement specifics to IHU.

It could be complemented and extended by adding new upwinding schemes such as Phase Upwind (PU) or weighted Hybrid Upwind (wIHU)

["Smooth implicit hybrid upwinding for compositional multiphase flow in porous media"(Bosma et al, 2022)](https://www.sciencedirect.com/science/article/pii/S0045782521005879)
["Upstream Differencing for Multiphase Flow in Reservoir Simulation"](https://epubs.siam.org/doi/10.1137/0728036)

### CODE REVIEW

**Review by paveltomin — APPROVED** [review]

File: src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVMUpwindUtilities.hpp [hunk 1]
```diff
@@ -0,0 +1,1597 @@
...
+    {
+      dFractionalFlow_dP[ke] = 0;
+      for( localIndex jc = 0; jc < numComp; ++jc )
+      {
+        dFractionalFlow_dC[ke][jc] = 0;
+      }
+    }
+
+    //Form totMob
+    for( localIndex jp = 0; jp < numPhase; ++jp )
```
[the remark this answers is not in the document]
Comment (jafranc) [reply src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseFVMUpwindUtilities.hpp]: Actually made me re-think it. Total mobility is now imported through interface and only assemble once.

**Comment by paveltomin** [comment]
@jafranc any chance to get it merged? really don't want to loose your efforts

**Comment by jafranc** [comment]
> @jafranc any chance to get it merged? really don't want to loose your efforts

Was in the dev of dispersion. Will do asa I have #2525 and #2600 merged as they are smaller (and #2600 is of interest for this one)

**Review by paveltomin — APPROVED** [review]
thanks!

**Comment by paveltomin** [comment]
@jafranc will you have time to merge it?

**Comment by jafranc** [comment]
> @jafranc will you have time to merge it?

It is plan for early march as needed for other projects

**Review by CusiniM — APPROVED** [review]

**Comment by CusiniM** [comment]
this does not seem to require baselines.

**Pull request merged**